lib/spreewald/email_steps.rb

```diff
@@ -88,34 +113,34 @@
 end.overridable
 
 # Checks that the last sent email includes some text
-Then /^I should see "([^\"]*)" in the e?mail$/ do |text|
-  expect(MailFinder.email_text_body(ActionMailer::Base.deliveries.last)).to include(text)
+Then /^I should see "([^\"]*)" in the( HTML| plain-text|) e?mail$/ do |text, type|
+  expect(MailFinder.email_text_body(ActionMailer::Base.deliveries.last, type.strip)).to include(text)
 end.overridable
 
-# Print all sent emails to STDOUT.
-Then /^show me the e?mails$/ do
-  ActionMailer::Base.deliveries.each_with_index do |mail, i|
-    puts "E-Mail ##{i}"
-    print "-" * 80
-    puts [ "From:    #{mail.from}",
-           "To:      #{mail.to}",
-           "Subject: #{mail.subject}",
-           "\n" + MailFinder.email_text_body(mail)
-         ].join("\n")
-    print "-" * 80
+# Print all sent emails to STDOUT (optionally only the headers).
+Then /^show me the e?mail( header)?s$/ do |only_header|
+  print MailFinder.show_mails(ActionMailer::Base.deliveries, only_header)
+  puts "No emails found" if ActionMailer::Base.deliveries.empty?
```

Comment:
It seems you are using `print` just to avoid the linebreak when there are no emails. Your code should deal with that instead.

```suggestion
  if ActionMailer::Base.deliveries.empty?
    puts "No emails found"
  else
    puts MailFinder.show_mails(ActionMailer::Base.deliveries, only_header)
  end
```